blueocean-dashboard/src/main/js/components/Capability.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import update from 'react-addons-update';
+import { classMetadataStore } from '@jenkins-cd/js-extensions';
+
+export const capabilityStore = classes => ComposedComponent => class extends Component {
```

Comment:
Seconded, I've actually never seen this kind of class declaration before and have no idea what it does :/ Some thorough jsdocs here would be great.